### DISCUSSION TITLE
docker compose uses arcane external volumes

### DISCUSSION THREAD

**Discussion by devkral** [dymmond/esmerald#313] [General]
the docker-compose files shouldn't contain references to external volumes.
If you want to persist the changes, a docker-compose.override.yml can be used instead.

It is hard to use for external contributors (this is why I removed this in my PRs)

**Comment by tarsil** [comment]
I haven't read anything in the official docker documentation about being arcane?

**Comment by tarsil** [comment]
Addressed now :)

**Comment by devkral** [comment]
the override is incorrect, it overrides everything and reintroduces the problem.
Have a look in my PR.
I meant to exclude the override in gitignore and to use it only locally where the volume exists

**Comment by devkral** [comment]
sry was an other problem.
Anyway it is override not overrides, so it works (not detected by docker). Otherwise it would fail

**Comment by tarsil** [comment]
Yes, I made a typo too @devkral I knew that bit. 

Just one quick question for clarification. I like hatch in so many levels, mostly because it creates the venvs for us and so on but something I forgot, literally, what if I want to install all the packages (test, doc, dev...) in one go? Could you remind me of the command?

**Reply by devkral** [reply]
something like an extra like "all" and 
in tests a dependency:

"esmerald[all]"   #  this includes the extra all of esmerald

would be possible.

Note:
I would exclude always the docs stuff. It simply has no value to be installed besides when building docs

**Reply by devkral** [reply]
a workaround would be to iterate through all environments and to output the requirements per environment via pip freeze

**Reply by devkral** [reply]
e.g.: `hatch -e test run pip freeze`
